sdk/documentintelligence/ai-document-intelligence-rest/src/documentIntelligence.ts

```diff
@@ -63,5 +65,39 @@ export default function createClient(
     },
   });
 
+  client.pipeline.addPolicy({
+    name: "ResponseStreamToUint8ArrayPolicy",
+    sendRequest: async (req, next) => {
+      logger.info("ResponseStreamToUint8ArrayPolicy starts");
+
+      const figuresUrlPattern = /\/documentModels\/[^/]+\/analyzeResults\/[^/]+\/figures\/[^/]+/;
+      if (req.method === "GET" && figuresUrlPattern.test(req.url)) {
+        logger.info(`Matched URL pattern for figures: ${req.url}`);
+        req.streamResponseStatusCodes = new Set([200]);
+        logger.info("Set streamResponseStatusCodes to 200 for the figures API");
+      }
+
+      const response = await next(req);
+
+      if (req.method === "GET" && figuresUrlPattern.test(req.url)) {
+        logger.info(`Handling response for figures URL: ${req.url}`);
+        if (!response.readableStreamBody) {
+          logger.error("Expected response to have a readable stream but found none");
+          throw new RestError("Expected response to have a readable stream", {
+            code: RestError.PARSE_ERROR,
+          });
+        }
+        logger.info("Converting readable stream to Uint8Array");
+        // Casting Uint8Array as NodeJS.ReadableStream which gets passed on as response.body, this lets users consume Uint8Array response body directly
+        response.readableStreamBody = (await streamToUint8Array(
```

Comment:
I feel like this should be done explicitly by the user (they use a helper we export to turn the stream into a Uint8Array) or else this will become a very brittle contract that could easily break when core-client-rest decides to become smarter about handling response bodies of different types.

Alternatively, perhaps we could add support to `core-client-rest` to do this handling directly, e.g. have a request flag that reads in the stream and assigns a Uint8Array to the body property. @joheredi do you have any thoughts here?

Reply:
We have quite a few RLC responses in the repo so a common support would be great!

Reply:
ok, going with exporting `streamToUint8Array` and letting users do their thing once they get the response.

Reply:
I also wonder whether other packages also have the same issue

Reply:
And that would mean users having to cast it..
```js
const output = await streamToUint8Array(response.body as unknown as NodeJS.ReadableStream)
fs.promises.writeFile(`./figures/${figureId}.png`, output);
```
is that ok?

Reply:
> I also wonder whether other packages also have the same issue

Looks that they are using `asNodeStream()`/`asBrowserStream` to get the body as stream even though their body is typed as `Uint8Array`.  https://github.com/Azure/azure-sdk-for-js/blob/39444d190ad501e48ef423a0ad0c281e0b697078/sdk/maps/maps-render-rest/samples-dev/getMapTileInNode.ts#L48

> Alternatively, perhaps we could add support to `core-client-rest` to do this handling directly, e.g. have a request flag that reads in the stream and assigns a Uint8Array to the body property. @joheredi do you have any thoughts here?

maybe similar to `asXxxStream()` we also provide `asBytes`?  Still, it requires customers to call these methods.

Reply:
They patched it that way?